src/scripts/main.js

```diff
@@ -1,3 +1,28 @@
 'use strict';
 
-// write code here
+const listEl = document.querySelector('ul');
+
+const convertToNumber = (str) => Number(str.replace('$', '').replace(',', ''));
+
+const sortList = (list, sortParam) =>
+  list.sort(
+    (a, b) =>
+      convertToNumber(b.dataset[sortParam]) - convertToNumber(a.dataset[sortParam]),
+  );
```

Comment:
The `sortList` function is currently mutating the original list because the `sort` method modifies the original array. This might lead to unexpected behavior in other parts of the code that use the same list. Consider creating a copy of the list before sorting it.